### PR TITLE
Display SonarDelphi errors instead of failing scan

### PR DESCRIPTION
**Pull request by fourls**
This PR refactors the handling of errors that occur within SonarDelphi during analysis. Previously, any error-level logging would cause the analysis to fail with a fatal error, which is inaccurate - SonarDelphi is quite resilient to errors and can conduct a successful partial analysis in most cases an error appears. This new approach communicates the logs directly to the client _along_ with any results, leaving the user free to determine how important the error is / continue using DelphiLint while they identify the problem.

A new "View Last Analysis Log" option is available in the "Analyze" dropdown:

image

Clicking this after an analysis shows a log window with options to save a log file or report an issue on integrated-application-development/sonar-delphi:

image

If the analysis produces errors, then the following shows up (clicking opens the log):

image

If the analysis produces warnings, then the following shows up (clicking opens the log):

image

Example of a log with errors:

image

### CODE REVIEW

File: client/source/DelphiLint.LogViewer.dfm [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    TabOrder = 0
+    object Label3: TLabel
+      Left = 8
+      Top = 6
+      Width = 718
+      Height = 15
+      Caption = 
+        'Please note that errors or warnings during an analysis can affect ' +
+        'the quality of the results, including false positive or false negative ' +
+        'issues.'
```
Comment (Cirras) [inline client/source/DelphiLint.LogViewer.dfm:32]: Is this a bit verbose? I'd have probably written it in a much more terse fashion like below, though I'm conscious that you want a consistent tone in the DelphiLint hints/messages.
```
Errors/warnings may affect analysis quality, causing missing issues or false positives.
```

File: client/source/DelphiLint.LogViewer.dfm [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    end
+    object Label4: TLabel
+      Left = 8
+      Top = 24
+      Width = 592
+      Height = 15
+      Caption = 
+        'Errors are usually caused by attempting to analyze an invalid file ' +
+        'or by misconfiguring a sonar-project.properties.'
```
Comment (Cirras) [inline client/source/DelphiLint.LogViewer.dfm:41]: Similar to above.
```
Errors are typically caused by malformed source files, or a misconfigured project.
```

File: client/source/DelphiLint.ToolFrame.dfm [hunk 3]
```diff
@@ -247,11 +247,75 @@ object LintToolFrame: TLintToolFrame
...
+      ParentBackground = False
+      TabOrder = 1
+      StyleElements = [seFont, seBorder]
+      object ErrorButton: TSpeedButton
+        Left = 0
+        Top = 0
+        Width = 132
+        Height = 27
+        Align = alClient
+        Caption = '2 errors'
```
Comment (Cirras) [inline client/source/DelphiLint.ToolFrame.dfm:267]: Why is the caption "2 errors" by default?

File: client/source/DelphiLint.ToolFrame.dfm [hunk 4]
```diff
@@ -264,10 +328,292 @@ object LintToolFrame: TLintToolFrame
     object ActionClearActiveFile1: TMenuItem
       Action = PluginCore.ActionClearActiveFile
     end
+    object Separator2: TMenuItem
+      Caption = '-'
+    end
+    object ViewLogItem: TMenuItem
+      Caption = '&View Last Analysis Log'
+      OnClick = ErrorButtonClick
```
Comment (Cirras) [inline client/source/DelphiLint.ToolFrame.dfm:336]: This event handler name references the error button but is also used by the `WarningButton` and `ViewLogItem`. Should the name be something else?

**Review by Cirras — CHANGES_REQUESTED** [review]
Great change.

Surfacing errors and warnings in this way is a nice win for consistency with standard SonarDelphi scans, and much less disruptive when things do go wrong.
I also really like the Analysis Log UI and hope the `Report a bug on GitHub` button will drive some useful bug reports to the upstream SonarDelphi project.

Testing went well.
- Should we have more visual feedback on the `Errors` and `Warnings` buttons?
  - The little magnifying glass is cute, but easy to miss. 
  - Tooltip on hover saying "View analysis log" or similar
  - Background color change on hover, like with other buttons.
- The GitHub icon in the Analysis Log is a nice touch, but there's some white pixels around it that look a little funny. BMP antialiasing issue?
  image



Code changes are mostly good to go with a few nitpicks, see below.

**Comment by fourls** [comment]
(rebase)

File: client/source/DelphiLint.ToolFrame.dfm (hunk 3 above)
Comment (fourls) [reply client/source/DelphiLint.ToolFrame.dfm:267]: It's just something to look at at design-time. Do you think I should make it something else?

File: client/source/DelphiLint.ToolFrame.dfm (hunk 4 above)
Comment (fourls) [reply client/source/DelphiLint.ToolFrame.dfm:336]: Changed to `ViewLogClick`.

**Comment by fourls** [comment]
Updated now based on feedback.

Log description:

image

GitHub icon (dark theme):

image

GitHub icon (light theme):

image

> Should we have more visual feedback on the Errors and Warnings buttons?
> * The little magnifying glass is cute, but easy to miss.
> * Tooltip on hover saying "View analysis log" or similar
> * Background color change on hover, like with other buttons.

VCL is surprisingly restrictive with what it lets you do with buttons. I've added a tooltip, but the background color is totally dependent on the theme. In light mode, for example, it actually does darken the buttons:

light_mode_errors

There's not too much I can do without implementing a custom control. I don't think what we have now will be too much of a problem - it's so colourful and angry that I expect most people will click on it anyway.

**Review by Cirras — APPROVED** [review]
Looks good to me. 🎉

File: client/source/DelphiLint.ToolFrame.dfm (hunk 3 above)
Comment (Cirras) [reply client/source/DelphiLint.ToolFrame.dfm:268]: Nah, that seems fine.

**Pull request merged**